src/main/java/edu/cornell/kfs/module/ar/document/service/impl/CuContractsGrantsInvoiceDocumentServiceImpl.java

```diff
@@ -59,6 +62,40 @@
                     totalCostInvoiceDetail.getTotalAmountBilledToDate().add(document.getInvoiceGeneralDetail().getCostShareAmount()));
         }
         
+        Award award = (Award) document.getInvoiceGeneralDetail().getAward();
+        if (ObjectUtils.isNotNull(award)) {
+            AwardExtendedAttribute awardExtension = (AwardExtendedAttribute) award.getExtension();
+            if (ObjectUtils.isNotNull(awardExtension)) {
+                localParameterMap.put(CuArPropertyConstants.ContractsAndGrantsBillingAwardFields.AWARD_BUDGET_START_DATE, 
+                        convertDateToString(awardExtension.getBudgetBeginningDate()));
+                localParameterMap.put(CuArPropertyConstants.ContractsAndGrantsBillingAwardFields.AWARD_BUDGET_END_DATE, 
+                        convertDateToString(awardExtension.getBudgetEndingDate()));
+                localParameterMap.put(CuArPropertyConstants.ContractsAndGrantsBillingAwardFields.AWARD_BUDGET_TOTAL, 
+                        awardExtension.getBudgetTotalAmount());
+                localParameterMap.put(CuArPropertyConstants.ContractsAndGrantsBillingAwardFields.AWARD_PRIME_AGREEMENT_NUMBER,
+                        convertNullStringToEmptyString(awardExtension.getPrimeAgreementNumber()));
+                localParameterMap.put(CuArPropertyConstants.ContractsAndGrantsBillingAwardFields.AWARD_PURCHASE_ORDER,
+                        convertNullStringToEmptyString(awardExtension.getPurchaseOrderNumber()));
+            }
+        }
+        
+        List<InvoiceAccountDetail> invoiceAccountDetails = document.getAccountDetails();
+        if (ObjectUtils.isNotNull(invoiceAccountDetails)) {
+            Account contractControlAccount = determineContractControlAccount(invoiceAccountDetails.get(0));
+            if (ObjectUtils.isNotNull(contractControlAccount)) {
+                Map<String, Object> primaryKeys = new HashMap<String, Object>();
+                primaryKeys.put(KFSPropertyConstants.PROPOSAL_NUMBER, invoiceAccountDetails.get(0).getProposalNumber());
+                primaryKeys.put(KFSPropertyConstants.CHART_OF_ACCOUNTS_CODE,contractControlAccount.getChartOfAccountsCode());
+                primaryKeys.put(KFSPropertyConstants.ACCOUNT_NUMBER, contractControlAccount.getAccountNumber());
+                AwardAccount awardAccount = businessObjectService.findByPrimaryKey(AwardAccount.class, primaryKeys);
+                AwardAccountExtendedAttribute awardAccountExtension = (AwardAccountExtendedAttribute) awardAccount.getExtension();
```

Comment:
should this be surrounded by a null check of awardAccount, since we have to look it up?

Reply:
Good catch.